content/shmem_test_some.tex

```diff
@@ -0,0 +1,85 @@
+\apisummary{
+  Test an array of variables on the local \ac{PE} to meet a specified test condition.
+}
+
+\begin{apidefinition}
+
+\begin{C11synopsis}
+size_t @\FuncDecl{shmem\_test\_some}@(TYPE *ivars, size_t nelems, size_t *indices, int *status,
+    int cmp, TYPE cmp_value);
+\end{C11synopsis}
+where \TYPE{} is one of the point-to-point synchronization types specified by
+Table \ref{p2psynctypes}.
+
+\begin{Csynopsis}
+size_t @\FuncDecl{shmem\_\FuncParam{TYPENAME}\_test\_some}@(TYPE *ivars, size_t nelems, size_t *indices,
+    int *status, int cmp, TYPE cmp_value);
+\end{Csynopsis}
+where \TYPE{} is one of the point-to-point synchronization types and has a
+corresponding \TYPENAME{} specified by Table \ref{p2psynctypes}.
+
+\begin{apiarguments}
+
+  \apiargument{OUT}{ivars}{A pointer to an array of remotely accessible data
+    objects.}
+  \apiargument{IN}{nelems}{The number of elements in the \VAR{ivars} array.}
+  \apiargument{OUT}{indices}{An array of indices of length at least
+    \VAR{nelems} into \VAR{ivars} that satisfied the test condition.}
+  \apiargument{INOUT}{status}{An optional mask array of length \VAR{nelems}
+    that indicates which elements in \VAR{ivars} are excluded from the test set.}
+  \apiargument{IN}{cmp}{A comparison operator from Table~\ref{p2p-consts}
+    that compares elements of \VAR{ivars} with \VAR{cmp\_value}.}
+  \apiargument{IN}{cmp\_value}{The value to be compared with the objects
+    pointed to by \VAR{ivars}.}
+
+\end{apiarguments}
+
+\apidescription{
+    The \FUNC{shmem\_test\_some} routine behaves similarly to
+    \FUNC{shmem\_wait\_until\_some}, but it does not block and returns zero if
+    no conditions are satisfied.  This routine tests the numeric comparison of
+    each of the \VAR{nelems} elements in the \VAR{ivars} array with the value
+    \VAR{cmp\_value} according to the comparison operator \VAR{cmp} at the
+    calling PE.  The order in which these elements are tested is unspecified.
+
+    Upon return, the \VAR{indices} array contains the indices of the elements
+    in the test set that satisfied the test condition during the call to
+    \FUNC{shmem\_test\_some}.  The return value of \FUNC{shmem\_test\_some} is
+    equal to the total number of these satisfied elements.  If the return value
+    is $N$, then the first $N$ elements of the \VAR{indices} array contain
+    those unique indices that satisfied the test condition.  The array pointed
+    to by indices must be at least \VAR{nelems} long.
+
+    The optional \VAR{status} array passed to \FUNC{shmem\_test\_some} has the
```

Comment:
Even though the rules around status are explained in shmem_wait_until_some, my personal preference would be to have them repeated in full here so that this API definition is self-contained.

Reply:
@grippa Yeah, I understand.  It's funny - I've also received the exact opposite critique, which is how this shorter version came to be :wink:.  Perhaps we can defer this decision to the doc edit process, since copy/pasting the description of `status` wouldn't change any semantics?

Reply:
Heh looks like I made this same comment previously. I'm still in favor of my original opinion.